### DISCUSSION TITLE
Handling dynamic properties on `WC_Order` instances with PHP 8.1+

### DISCUSSION THREAD

**Discussion by Sidsector9** [woocommerce/woocommerce#43367] [General]
We're working on a very large WooCommerce extension and dealing with very old code, where dynamic properties have been added to the order object returned by `wc_get_order()`. 

Dynamic properties are deprecated since PHP 8.1. We've tried fixing it using `WC_Order_Wrapper extends WC_Order {}` and declaring the dynamic properties inside `WC_Order_Wrapper`, and then using it as:

`$order = new WC_Order_Wrapper( $order );` but this fails with subscriptions, and our plugin works with subscription products.

And from what I understand, using the `woocommerce_order_class` filter hook will cause the same issue with subscription orders as well.

What would be the correct way to handle this situation?

**Comment by lsinger** [comment]
Hey @Sidsector9, thanks for opening up this discussion. Could you provide us with some minimal example code to reproduce the warning that you’re seeing?

**Comment by KratosGemini** [comment]
I'm not very familiar with customizing this part of WooCommerce, but I opened an issue with the WooCommerce Square plugin regarding these types of deprecation warnings. Perhaps what comes of that will be helpful. I don't know what the "WooCommerce way" is when it comes to this. The only thing I've ventured into is using the 'woocommerce_checkout_create_order_line_item' action to add custom meta data to an order.

https://github.com/woocommerce/woocommerce-square/issues/219
